models/users.js

```diff
@@ -260,7 +260,11 @@ const fetchUser = async ({ userId = null, username = null, githubUsername = null
         userData = doc.data();
       });
     } else if (discordId) {
-      const user = await userModel.where("discordId", "==", discordId).limit(1).get();
+      const user = await userModel
+        .where("discordId", "==", discordId)
+        .where("roles.in_discord", "==", true)
+        .limit(1)
+        .get();
```

Comment:
Does this solve everything mentioned over the issue

Reply:
Yes it does.